Give access to stats returned by server in X-ClickHouse-Summary

bgranvea opened the pull request:
Clickhouse can return interesting statistics in a X-ClickHouse-Summary header if send_progress_in_http_headers parameter is set (see https://github.com/ClickHouse/ClickHouse/pull/5116). 

In my use case, I need this to know how many rows are inserted when executing a INSERT ... SELECT ...

pavelnemirovsky commented:
@bgranvea thanks for PR, any plan to fix ci checks?

bgranvea commented:
> @bgranvea thanks for PR, any plan to fix ci checks?

there was an issue not related to this PR, I think it is fixed by https://github.com/ClickHouse/clickhouse-jdbc/commit/4188aafc52d4eef4ad416154fc2d1f431e0cb941

I've also added the possibility to add query param before executing it, with the use of ConfigurableApi as it is done in Writer, because I need this in my use case.

PR is updated but I don't see the CI results anymore? is there something to do to rerun the tests?

den-crane commented:
Tests are OK https://travis-ci.org/github/ClickHouse/clickhouse-jdbc/pull_requests
But it does not reflected in the PR for some  reason.

zhicwu commented on src/test/java/ru/yandex/clickhouse/integration/ResultSummary.java:
```diff
@@ -0,0 +1,130 @@
+import static org.testng.Assert.assertNull;
+import static org.testng.AssertJUnit.assertTrue;
+
+public class ResultSummary {
```
Could you rename the class to `ResultSummaryTest` for consistency?

zhicwu commented on src/test/java/ru/yandex/clickhouse/integration/ResultSummary.java:
```diff
@@ -0,0 +1,130 @@
+
+    @BeforeTest
+    public void setUp() throws Exception {
+        ClickHouseProperties properties = new ClickHouseProperties();
+        ClickHouseDataSource dataSource = new ClickHouseDataSource("jdbc:clickhouse://localhost:8123", properties);
+        connection = dataSource.getConnection();
```
As we're now using [TestContainer](https://www.testcontainers.org/) to eliminate hard-coded connection string and improved CI experience, could you change these 3 lines to below?
```java
connection = ClickHouseContainerForTest.newDataSource().getConnection();
```

zhicwu commented:
Great work @bgranvea! Do you have time to update the PR and re-target to `develop` branch? I'm wondering if we can make it part of 0.2.6 release aiming Feb 13th.

bgranvea commented:
> Great work @bgranvea! Do you have time to update the PR and re-target to `develop` branch? I'm wondering if we can make it part of 0.2.6 release aiming Feb 13th.

I've updated with your remarks and rebased on the develop branch.

zhicwu commented:
> I've updated with your remarks and rebased on the develop branch.

Thank you @bgranvea. Could you change merge target from master to develop?

den-crane commented:
BTW, Number of affected rows will be inaccurate. 
CH takes into account rows inserted into Materialized Views so the number will be bigger.
And the latest CH version does merge during insert so the the number can be fewer.

bgranvea commented:
> BTW, Number of affected rows will be inaccurate.
> CH takes into account rows inserted into Materialized Views so the number will be bigger.
> And the latest CH version does merge during insert so the the number can be fewer.

I also noticed that with distributed tables the number of inserted rows is doubled.
